### PR TITLE
Remove reverse dependency to webconsole module

### PR DESCRIPTION
**Pull request by neumayer**
Both initialisation and shutdown hook of the webconsole are handled via
the module interface now. The SecondBase class will call init methods of
all modules and also check if they provide custom shutdown hooks. If
they do it will be added at runtime.

### CODE REVIEW

File: core/src/main/java/org/secondbase/core/SecondBase.java [hunk 1]
```diff
@@ -56,27 +55,11 @@ private void init(final String[] args, final Flags flags) throws SecondBaseExcep
...
         }
 
-        if (webConsole != null) {
-            try {
-                webConsole.start();
-            } catch (final IOException e) {
-                throw new SecondBaseException("Could not start webconsole.", e);
+        for (final SecondBaseModule module : configurableModule) {
+            if(module.getShutdownHook().isPresent()) {
+                Runtime.getRuntime().addShutdownHook(module.getShutdownHook().get());
```
Comment (acidmoose) [inline core/src/main/java/org/secondbase/core/SecondBase.java]: Can't the module itself add the shutdown hook? Why does it need to be coordinated at all? Consul, for instance, has it's own shutdown hook now.

**Review by acidmoose — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/secondbase/core/SecondBase.java (hunk 1 above)
Comment (neumayer) [reply core/src/main/java/org/secondbase/core/SecondBase.java]: Yes, you are right, it can, I made it.

Comment (neumayer) [reply core/src/main/java/org/secondbase/core/SecondBase.java]: Done.

File: webconsole/src/main/java/org/secondbase/webconsole/HttpWebConsole.java [hunk 2]
```diff
@@ -61,8 +63,23 @@ public void load(final SecondBase secondBase) {
...
-    public void init() {
-        // nothing to initialise
+    public void init() throws SecondBaseException {
+        try {
+            start();
+        } catch (final IOException e) {
+            throw new SecondBaseException("Could not start webconsole.", e);
+        }
+
+                Runtime.getRuntime().addShutdownHook(new Thread() {
```
Comment (acidmoose) [inline webconsole/src/main/java/org/secondbase/webconsole/HttpWebConsole.java]: Indenting?

**Review by acidmoose — CHANGES_REQUESTED** [review]

**Review by acidmoose — APPROVED** [review]

**Pull request merged**